lib/ione/io/io_reactor.rb

```diff
@@ -138,20 +138,19 @@ def running?
       def start
         @lock.synchronize do
           if @state == RUNNING_STATE
-            return @started_promise.future
+            return @started_promise
           elsif @state == STOPPING_STATE
-            return @stopped_promise.future.flat_map { start }.fallback { start }
+            return @stopped_promise.then { |_| start }.rescue { start }
```

Comment:
I think this is wrong. Should be `flat`. https://github.com/ruby-concurrency/concurrent-ruby/blob/master/docs-source/promises.in.md#flatting